libs/langchain-anthropic/src/chat_models.ts

```diff
@@ -766,6 +782,30 @@ export class ChatAnthropicMessages<
       | Anthropic.Messages.ToolChoiceTool
       | undefined = handleToolChoice(options?.tool_choice);
 
+    if (this.thinking.type === "enabled") {
+      if (this.topK !== -1) {
+        throw new Error("topK is not supported when thinking is enabled");
+      }
+      if (this.topP !== -1) {
+        throw new Error("topP is not supported when thinking is enabled");
+      }
+      if (this.temperature !== 1) {
+        throw new Error(
+          "temperature is not supported when thinking is enabled"
+        );
+      }
```

Comment:
we're using magic values for these rather than leaving them `undefined` in the instance vars, so I had to drop them here, and I didn't want to drop them silently.